File: FieldDocumentMaker.Editor/src/state/fields/field-model.ts
export type FieldTypes = 'Image' | 'Text' | 'Integer' | 'Float' | 'Date' | 'Combo'

export interface FieldBind {
    style: string
    bind: string
}

export interface FieldBase extends FieldBind {
    label: string
    value: string
}

export interface FieldModel {
    base: FieldBase
    type: FieldTypes
}

export interface TextField extends FieldModel {
    text: string
}

export interface NumberField extends FieldModel {
    number: number
    maxValue: number
    minValue: number
}

export interface DateField extends FieldModel {
    date: number
    maxDate: number
    minDate: number
}

export interface ComboField extends FieldModel {
    class: string
}

export interface ImageField extends FieldModel {
    url: string
    width: number
    height: number
}

File: FieldDocumentMaker.WPF/Window/WebBrowser/EditorModels/FieldModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FieldDocumentMaker.WPF.Window.WebBrowser.EditorModels
{
    class FieldBase
    {
        public string label { get; set; }

        public string value { get; set; }

        public string style { get; set; }

        public string bind { get; set; }

    }

    class FieldModel 
    {
        public FieldBase @base { get; set; }

        public string type { get; set; }
    }

    class TextField : FieldModel
    {
        public string text { get; set; }
    }

    class NumberField<T> : FieldModel
    {
        public T number { get; set; }
        public T maxValue { get; set; }
        public T minValue { get; set; }
    }

    class DateField : FieldModel
    {
        public DateTime date { get; set; }
        public DateTime maxDate { get; set; }
        public DateTime minDate { get; set; }
    }

    class ComboField : FieldModel
    {
        public string @class { get; set; }
    }

    class ImageField : FieldModel
    {
        public string url { get; set; }
        public int with { get; set; }
        public int height { get; set; }
    }
}
